### PR TITLE
Discount Module

### PR DESCRIPTION
**Pull request by UtsavSomaiya**

### CODE REVIEW

File: custom.js [hunk 1]
```diff
@@ -53,9 +53,16 @@ function displayCart() {
...
+    digit = document.getElementById('discount-selection').value.trim();
     let discount = new Array(cart.length);
     let tax = new Array(cart.length);
+    if (digit.substr(digit.length - 1) == "%") {
+        discountType = 1;
+    } else {
+        discountType = 2;
+    }
+    digit = parseInt(digit.replace(/[$%]/, ''));
+    console.log(digit);
```
Comment (hirenkeraliya) [inline custom.js:65]: ?

File: custom.js [hunk 2]
```diff
@@ -207,4 +214,23 @@ function removeFromCart(productId) {
         cart.splice(indexOfProduct, 1);
         displayCart();
     }
-}
\ No newline at end of file
+}
+
+function toggleModal(modalID) {
```
Comment (hirenkeraliya) [inline custom.js:219]: which modal?

File: custom.js [hunk 3]
```diff
@@ -207,4 +214,23 @@ function removeFromCart(productId) {
...
+    document.getElementById(modalID).classList.toggle("hidden");
+    document.getElementById(modalID + "-backdrop").classList.toggle("hidden");
+    document.getElementById(modalID).classList.toggle("flex");
+    document.getElementById(modalID + "-backdrop").classList.toggle("flex");
+}
+
+function toast() {
+    var x = document.getElementById("snackbar");
```
Comment (hirenkeraliya) [inline custom.js:227]: what is x?

File: discount.php [hunk 4]
```diff
@@ -0,0 +1,49 @@
+<div class="hidden overflow-x-hidden overflow-y-auto fixed inset-0 z-50 outline-none focus:outline-none justify-center items-center" id="modal-id">
+    <div class="relative w-auto my-6 mx-auto max-w-sm">
+        <div class="border-0 rounded-lg shadow-lg relative flex flex-col w-full bg-white outline-none focus:outline-none">
+            <div class="flex items-start justify-between p-5 border-b border-solid border-blueGray-200 rounded-t">
+                <h3 class="text-3xl font-semibold">
+                        Discounts
```
Comment (hirenkeraliya) [inline discount.php:6]: one tab back

**Review by hirenkeraliya — CHANGES_REQUESTED** [review]

File: custom.js [hunk 5]
```diff
@@ -53,9 +53,22 @@ function displayCart() {
...
-    digit = document.getElementById('discount-digit').innerHTML.trim();
+    discountId = document.getElementById('discount-selection').value.trim();
+    discountDigit = document.getElementById('discount-option-' + discountId).innerHTML.trim();
     let discount = new Array(cart.length);
     let tax = new Array(cart.length);
+    if (discountDigit.substr(discountDigit.length - 1) == "%") {
+        discountType = 1;
+    } else {
+        discountType = 2;
+    }
```
Comment (hirenkeraliya) [inline custom.js:64]: Do not need else part

File: custom.js [hunk 6]
```diff
@@ -207,4 +220,23 @@ function removeFromCart(productId) {
         cart.splice(indexOfProduct, 1);
         displayCart();
     }
-}
\ No newline at end of file
+}
+
+function discountModal(modalId) {
```
Comment (hirenkeraliya) [inline custom.js:225]: displayApplicableDiscountsModal

File: discount.php [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+                        ×
+                    </span>
+                </button>
+            </div>
+            <div class="relative p-6 flex-auto">
+                <div class="flex justify-center">
+                    <div class="mb-3 xl:w-96">
+                        <select class="form-select appearance-none block w-full px-3 py-1.5 text-base font-normal text-gray-700 bg-white bg-clip-padding bg-no-repeat border border-solid border-gray-300 rounded transition ease-in-out m-0 focus:text-gray-700 focus:bg-white focus:border-blue-600 focus:outline-none" aria-label="Default select example" onchange="discountApply()" id="discount-selection">
+                            <option value="">--Select discount--</option>
+                            <?php for ($i=0;$i<sizeof($discounts);$i++) {?>
```
Comment (hirenkeraliya) [inline discount.php:19]: Proper indentation

File: discount.php [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+                                                echo "selected";
+                                            }
+                                        ?> id="discount-option-<?= $discounts[$i]['id']?>">
+                                    <?php
+                                        if ($discounts[$i]['type'] == 1) {
+                                            echo $discounts[$i]['name']."(".$discounts[$i]['digit']."%)";
+                                        }
+                                        echo $discounts[$i]['name']."($".$discounts[$i]['digit'].")";
+                                    ?>
+                                </option>
```
Comment (hirenkeraliya) [inline discount.php:32]: Do some readable code

**Comment by UtsavSomaiya** [comment]
https://user-images.githubusercontent.com/96036522/157170640-0cc3bc48-3549-45fe-bca6-169a682a7ff0.MP4

**Review by hirenkeraliya — APPROVED** [review]

**Pull request merged**